app/assets/javascripts/discourse/app/lib/keyboard-shortcuts.js

```diff
@@ -782,6 +781,18 @@ export default {
     article.setAttribute("tabindex", "0");
     article.focus();
 
+    if (article.classList.contains("topic-list-item")) {
+      this.prefetchService.fetch(
+        article.dataset.topicId,
+        article.dataset.lastReadPostNumber
+      );
+    }
```

Comment:
Seems a bit weird to have this mixed up in keyboard shortcuts. How about making it happen during the `onfocus` event of a topic link?